File: src/main/scala/com/kirkwoodwest/scalawoods/utils/MidiUtil.scala
package com.kirkwoodwest.scalawoods.utils

/**
 * Some basic helpers for midi
 */
object MidiUtil {
  val NOTE_ON = 0x90
  val NOTE_OFF = 0x80
  val CC = 0xB0

  def random_127: Int = (Math.random * 127).toInt

  def doubleTo127(value: Double): Int = (value * 127).toInt
  
  def intToHex(value: Int): String = value.toHexString
}

class MidiUtil {
  def getStatusWithChannel(status: Int, channel: Int): Int = status | channel
}

File: src/main/java/com/kirkwoodwest/openwoods/utils/MidiUtil.java
package com.kirkwoodwest.openwoods.utils;

/**
 * Some basic helpers for midi
 */
public class MidiUtil {
  public static final int NOTE_ON = 0x90;
  public static final int NOTE_OFF = 0x80;
  public static final int CC   = 0xB0;
  public int getStatusWithChannel(int status, int channel){
    return status | channel;
  }
  public static int random_127() {
    return (int) (Math.random() * 127);
  };
  public static int doubleTo127(double value) {
    return (int) (value * 127);
  };
}
